packages/repository/src/repositories/relation.factory.ts

```diff
@@ -0,0 +1,149 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/example-todo
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {EntityCrudRepository} from '.';
+import {
+  Model,
+  RelationType,
+  AnyObject,
+  Options,
+  Entity,
+  Class,
+  FilterBuilder,
+  Filter,
+  WhereBuilder,
+  Where,
+  DataObject,
+} from '..';
+import {cloneDeep, isArray} from 'lodash';
+import {
+  HasManyEntityCrudRepository,
+  DefaultHasManyEntityCrudRepository,
+} from './relation.repository';
+
+export declare type RelationDefinitionBase = {
+  type: RelationType;
+  modelFrom: Class<Entity> | string;
+  keyTo: string;
+  keyFrom: string;
```

Comment:
`keyTo` and `keyFrom` need to be expanded to be `string[]` so that we can allow compound keys.